app/controllers/hosts_controller.rb

```diff
@@ -671,7 +671,7 @@ def process_hostgroup
     return head(:not_found) unless @hostgroup
 
     @host = if params[:host][:id]
-              host = Host::Base.authorized(:view_hosts, Host).find(params[:host][:id])
+              host = Host::Base.authorized(:view_hosts).find(params[:host][:id])
```

Comment:
This will work since new resource type will now be `Host::Base` and when we search for collection of filters, we translate it back to `Host` at https://github.com/sean797/foreman/blob/bc929bc30b277e907ceda208773932e6a2a92a37/app/services/authorizer.rb#L174

Since it's not clear at first look, it would be great to add a test in which we'd create a new Host subclass and tested it with it. This way it would make sure it won't break discovery in future.